src/pages/profile/Profile.jsx

```diff
@@ -0,0 +1,216 @@
+  */
+
+    // default avatar png       
+    const [avatarUrl, setAvatarUrl] = useState(AddImg);
+    const [photoBinary, setPhotoBinary] = useState(null);
+    const [loading, setLoading] = useState(false);
+
+    // username state
+    const [userName, setUserName] = useState(currentUser.displayName);
+
+    const [changeData, setChangeData] = useState({
+        nickname: ""
+        //  changed data object
+    })
+
+    useEffect(() => {
+
+        if (currentUser?.photoURL) {
+            setAvatarUrl(currentUser.photoURL);
+        }
+        setUserName(currentUser.displayName);
+
+    }, [currentUser.photoURL, currentUser.displayName])
+
+
+    const handleChange = (e) => {
+        if (e.target.files[0]) {
+            setPhotoBinary(e.target.files[0]);
+        }
+    }
+
+    const handleUpload = () => {
+        handleAvatarUpload(photoBinary);
+        setPhotoBinary();
+    }
+
+    const handleAvatarUpload = async (file) => {
+        const fileRef = ref(storage, currentUser.uid + ".png");
+        setLoading(true);
+
+        const snapshot = uploadBytesResumable(fileRef, file)
+
+        // accessing the url of the photo
+        const photoURL = await getDownloadURL(fileRef)
+
+        updateProfile(currentUser, { photoURL });
+
+        setLoading(false);
+        alert("File was uploaded");
+
+    }
+
+    // nickname input functionality
+    const handleNameChange = (event) => {
+        const { name, value } = event.target;
+
+        setChangeData({
+            ...changeData,
+            [name]: value,
+        })
+        console.log(changeData.nickname)
+        // Still to add error validation
+    }
+
+    //Need to review the rerender of the name
+
+    const handleNameChangeSubmit = (event) => {
+        event.preventDefault();
+        updateProfile(currentUser, { displayName: changeData.nickname });
+        if (event.target[0].value) {
+            event.target[0].value = '';
+        }
+        console.log(currentUser)
+    }
+
+    /*
+    User auth functionality
+    */
+
+    const [emailAuth, setEmailAuth] = useState('');
+    const [errorEmailAuth, setErrorEmailAuth] = useState('');
+
+    const validateEmailField = (value) => {
+
+        if (!value.trim()) {
+            return 'Email is required';
+        } else if (!/\S+@\S+\.\S+/.test(value)) {
+            return 'Email is invalid';
+        }
+        return '';
+    };
+
+
+    const handleEmailAuthChange = (event) => {
+        const value = event.target.value;
+        setEmailAuth(value);
+        //after user input validate that field matches the requirements
+
+        setErrorEmailAuth(validateEmailField(value));
+    };
+
+
+    const handlePassResetSubmit = (event) => {
+        event.preventDefault();
+        const newError = validateEmailField(emailAuth);
+
+        setErrorEmailAuth(newError);
+
+        if (newError.length === 0) {
+            // call submit function
+            sendPasswordResetEmail(auth, emailAuth)
```

Comment:
Why send resetEmail? If user reached this page he never forgot password, just want to change it